internal/server/handlers/rbac/database_engine_test.go

```diff
@@ -0,0 +1,581 @@
+						}) &&
+						slices.ContainsFunc(list.Items, func(engine everestv1alpha1.DatabaseEngine) bool {
+							return engine.GetName() == common.PGOperatorName
+						}) &&
+						slices.ContainsFunc(list.Items, func(engine everestv1alpha1.DatabaseEngine) bool {
+							return engine.GetName() == common.PSMDBOperatorName
+						})
+				},
+			},
+			{
+				desc: "read only pxc in default namespaces",
+				policy: newPolicy(
+					"p, role:test, database-engines, read, default/percona-xtradb-cluster-operator",
+					"g, bob, role:test",
+				),
+				assert: func(list *everestv1alpha1.DatabaseEngineList) bool {
+					return len(list.Items) == 1 &&
+						slices.ContainsFunc(list.Items, func(engine everestv1alpha1.DatabaseEngine) bool {
+							return engine.GetName() == common.PXCOperatorName
+						})
+				},
+			},
+			{
+				desc: "read only pg in default namespaces",
+				policy: newPolicy(
+					"p, role:test, database-engines, read, default/percona-postgresql-operator",
+					"g, bob, role:test",
+				),
+				assert: func(list *everestv1alpha1.DatabaseEngineList) bool {
+					return len(list.Items) == 1 &&
+						slices.ContainsFunc(list.Items, func(engine everestv1alpha1.DatabaseEngine) bool {
+							return engine.GetName() == common.PGOperatorName
+						})
+				},
+			},
+			{
+				desc: "read only psmdb in default namespaces",
+				policy: newPolicy(
+					"p, role:test, database-engines, read, default/percona-server-mongodb-operator",
+					"g, bob, role:test",
+				),
+				assert: func(list *everestv1alpha1.DatabaseEngineList) bool {
+					return len(list.Items) == 1 &&
+						slices.ContainsFunc(list.Items, func(engine everestv1alpha1.DatabaseEngine) bool {
+							return engine.GetName() == common.PSMDBOperatorName
+						})
+				},
+			},
+			{
+				desc: "read only psmdb & pg in default namespaces",
+				policy: newPolicy(
+					"p, role:test, database-engines, read, default/percona-server-mongodb-operator",
+					"p, role:test, database-engines, read, default/percona-postgresql-operator",
+					"g, bob, role:test",
+				),
+				assert: func(list *everestv1alpha1.DatabaseEngineList) bool {
+					return len(list.Items) == 2 &&
+						slices.ContainsFunc(list.Items, func(engine everestv1alpha1.DatabaseEngine) bool {
+							return engine.GetName() == common.PSMDBOperatorName
+						}) &&
+						slices.ContainsFunc(list.Items, func(engine everestv1alpha1.DatabaseEngine) bool {
+							return engine.GetName() == common.PSMDBOperatorName
+						})
+				},
+			},
+			{
+				desc: "read only psmdb in all namespaces",
+				policy: newPolicy(
+					"p, role:test, database-engines, read, */percona-server-mongodb-operator",
+					"g, bob, role:test",
+				),
+				assert: func(list *everestv1alpha1.DatabaseEngineList) bool {
+					return len(list.Items) == 1 &&
+						slices.ContainsFunc(list.Items, func(engine everestv1alpha1.DatabaseEngine) bool {
+							return engine.GetName() == common.PSMDBOperatorName
+						})
+				},
+			},
```

Comment:
I think we should add a negative case for a namespace, e.g.:
```suggestion
			},
			{
				desc: "read all in kube-system namespace",
				policy: newPolicy(
					"p, role:test, database-engines, read, kube-system/*",
					"g, bob, role:test",
				),
				assert: func(list *everestv1alpha1.DatabaseEngineList) bool {
					return len(list.Items) == 0
				},
			},
```